TF 2.0.0 Python 3.8 TypeError: _logger_find_caller() takes from 0 to 1 positional arguments but 2 were given

**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):  See script from Tensorflow training session and uploaded file below.  Nb: There is no error with TF2.0.0 and python 3.6 or 3.7.  The error occurs with TF2.0.0 and python 3.8.
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Ubuntu 18.04
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 2.0.0
- Python version: 3.8
- Bazel version (if compiling from source): 0.26.1
- GCC/Compiler version (if compiling from source): 7.4.0
- CUDA/cuDNN version: CUDA 10/cuDNN 7.6.4
- GPU model and memory: NVidia RTX 2080 TI and 2080 MaxQ

**Describe the current behavior**

After running the code below (with the attached file), you get the following error:

---------------------------------------------------------------------------
AssertionError                            Traceback (most recent call last)
~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/impl/api.py in converted_call(f, args, kwargs, caller_fn_scope, options)
    525         options=options, autograph_module=tf_inspect.getmodule(converted_call))
--> 526     converted_f = conversion.convert(target_entity, program_ctx)
    527 

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/impl/conversion.py in convert(entity, program_ctx)
    324 
--> 325   converted_entity_info = _convert_with_cache(entity, program_ctx,
    326                                               free_nonglobal_var_names)

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/impl/conversion.py in _convert_with_cache(entity, program_ctx, free_nonglobal_var_names)
    238 
--> 239     nodes, converted_name, entity_info = convert_entity_to_ast(
    240         entity, program_ctx)

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/impl/conversion.py in convert_entity_to_ast(o, program_ctx)
    474   elif tf_inspect.ismethod(o):
--> 475     nodes, name, entity_info = convert_func_to_ast(o, program_ctx)
    476   elif hasattr(o, '__class__'):

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/impl/conversion.py in convert_func_to_ast(f, program_ctx, do_rename)
    672   context = converter.EntityContext(namer, entity_info, program_ctx, new_name)
--> 673   node = node_to_graph(node, context)
    674 

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/impl/conversion.py in node_to_graph(node, context)
    702   node = converter.standard_analysis(node, context, is_initial=True)
--> 703   node = converter.apply_(node, context, function_scopes)
    704   node = converter.apply_(node, context, arg_defaults)

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/core/converter.py in apply_(node, context, converter_module)
    408   node = standard_analysis(node, context)
--> 409   node = converter_module.transform(node, context)
    410   return node

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/converters/function_scopes.py in transform(node, ctx)
    119 def transform(node, ctx):
--> 120   return FunctionBodyTransformer(ctx).visit(node)

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/core/converter.py in visit(self, node)
    345     try:
--> 346       return super(Base, self).visit(node)
    347     finally:

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/pyct/transformer.py in visit(self, node)
    479     if not anno.hasanno(node, anno.Basic.SKIP_PROCESSING):
--> 480       result = super(Base, self).visit(node)
    481     self.ctx.current_origin = parent_origin

/usr/local/lib/python3.8/ast.py in visit(self, node)
    359         visitor = getattr(self, method, self.generic_visit)
--> 360         return visitor(node)
    361 

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/converters/function_scopes.py in visit_FunctionDef(self, node)
    101     """
--> 102     wrapped_body = templates.replace(
    103         template,

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/pyct/templates.py in replace(template, **replacements)
    268   for node in nodes:
--> 269     node = ReplaceTransformer(replacements).visit(node)
    270     if isinstance(node, (list, tuple)):

/usr/local/lib/python3.8/ast.py in visit(self, node)
    359         visitor = getattr(self, method, self.generic_visit)
--> 360         return visitor(node)
    361 

/usr/local/lib/python3.8/ast.py in generic_visit(self, node)
    435                     if isinstance(value, AST):
--> 436                         value = self.visit(value)
    437                         if value is None:

/usr/local/lib/python3.8/ast.py in visit(self, node)
    359         visitor = getattr(self, method, self.generic_visit)
--> 360         return visitor(node)
    361 

/usr/local/lib/python3.8/ast.py in generic_visit(self, node)
    444             elif isinstance(old_value, AST):
--> 445                 new_node = self.visit(old_value)
    446                 if new_node is None:

/usr/local/lib/python3.8/ast.py in visit(self, node)
    359         visitor = getattr(self, method, self.generic_visit)
--> 360         return visitor(node)
    361 

/usr/local/lib/python3.8/ast.py in generic_visit(self, node)
    435                     if isinstance(value, AST):
--> 436                         value = self.visit(value)
    437                         if value is None:

/usr/local/lib/python3.8/ast.py in visit(self, node)
    359         visitor = getattr(self, method, self.generic_visit)
--> 360         return visitor(node)
    361 

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/pyct/templates.py in visit_Name(self, node)
    199 
--> 200     new_nodes = self._prepare_replacement(node, node.id)
    201 

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/pyct/templates.py in _prepare_replacement(self, replaced, key)
    138 
--> 139     new_nodes = ast_util.copy_clean(repl, preserve_annos=self.preserved_annos)
    140     if isinstance(new_nodes, gast.AST):

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/pyct/ast_util.py in copy_clean(node, preserve_annos)
     75   """
---> 76   return CleanCopier(preserve_annos).copy(node)
     77 

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/pyct/ast_util.py in copy(self, node)
     53       if not f.startswith('__') and hasattr(node, f):
---> 54         new_fields[f] = self.copy(getattr(node, f))
     55     new_node = type(node)(**new_fields)

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/pyct/ast_util.py in copy(self, node)
     40     if isinstance(node, list):
---> 41       return [self.copy(n) for n in node]
     42     elif isinstance(node, tuple):

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/pyct/ast_util.py in <listcomp>(.0)
     40     if isinstance(node, list):
---> 41       return [self.copy(n) for n in node]
     42     elif isinstance(node, tuple):

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/pyct/ast_util.py in copy(self, node)
     54         new_fields[f] = self.copy(getattr(node, f))
---> 55     new_node = type(node)(**new_fields)
     56 

~/tf38/lib/python3.8/site-packages/gast/gast.py in create_node(self, *args, **kwargs)
      9         nbparam = len(args) + len(kwargs)
---> 10         assert nbparam in (0, len(Fields)), \
     11             "Bad argument number for {}: {}, expecting {}".\

AssertionError: Bad argument number for keyword: 1, expecting 2

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-10-8b26b7af23a7> in <module>
----> 1 tf_model.fit(Xs_train[:, 0:1], y_train.reshape(-1, 1));

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/eager/def_function.py in __call__(self, *args, **kwds)
    566         xla_context.Exit()
    567     else:
--> 568       result = self._call(*args, **kwds)
    569 
    570     if tracing_count == self._get_tracing_count():

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/eager/def_function.py in _call(self, *args, **kwds)
    613       # This is the first call of __call__, so we have to initialize.
    614       initializers = []
--> 615       self._initialize(args, kwds, add_initializers_to=initializers)
    616     finally:
    617       # At this point we know that the initialization is complete (or less

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/eager/def_function.py in _initialize(self, args, kwds, add_initializers_to)
    494     self._graph_deleter = FunctionDeleter(self._lifted_initializer_graph)
    495     self._concrete_stateful_fn = (
--> 496         self._stateful_fn._get_concrete_function_internal_garbage_collected(  # pylint: disable=protected-access
    497             *args, **kwds))
    498 

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/eager/function.py in _get_concrete_function_internal_garbage_collected(self, *args, **kwargs)
   2363       args, kwargs = None, None
   2364     with self._lock:
-> 2365       graph_function, _, _ = self._maybe_define_function(args, kwargs)
   2366     return graph_function
   2367 

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/eager/function.py in _maybe_define_function(self, args, kwargs)
   2671 
   2672       self._function_cache.missed.add(call_context_key)
-> 2673       graph_function = self._create_graph_function(args, kwargs)
   2674       self._function_cache.primary[cache_key] = graph_function
   2675       return graph_function, args, kwargs

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/eager/function.py in _create_graph_function(self, args, kwargs, override_flat_arg_shapes)
   2551     arg_names = base_arg_names + missing_arg_names
   2552     graph_function = ConcreteFunction(
-> 2553         func_graph_module.func_graph_from_py_func(
   2554             self._name,
   2555             self._python_function,

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/framework/func_graph.py in func_graph_from_py_func(name, python_func, args, kwargs, signature, func_graph, autograph, autograph_options, add_control_dependencies, arg_names, op_return_value, collections, capture_by_value, override_flat_arg_shapes)
    956                                           converted_func)
    957 
--> 958       func_outputs = python_func(*func_args, **func_kwargs)
    959 
    960       # invariant: `func_outputs` contains only Tensors, CompositeTensors,

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/eager/def_function.py in wrapped_fn(*args, **kwds)
    437         # __wrapped__ allows AutoGraph to swap in a converted function. We give
    438         # the function a weak reference to itself to avoid a reference cycle.
--> 439         return weak_wrapped_fn().__wrapped__(*args, **kwds)
    440     weak_wrapped_fn = weakref.ref(wrapped_fn)
    441 

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/eager/function.py in bound_method_wrapper(*args, **kwargs)
   3179     # However, the replacer is still responsible for attaching self properly.
   3180     # TODO(mdan): Is it possible to do it here instead?
-> 3181     return wrapped_fn(*args, **kwargs)
   3182   weak_bound_method_wrapper = weakref.ref(bound_method_wrapper)
   3183 

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/framework/func_graph.py in wrapper(*args, **kwargs)
    935           # TODO(mdan): Push this block higher in tf.function's call stack.
    936           try:
--> 937             return autograph.converted_call(
    938                 original_func,
    939                 args,

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/impl/api.py in converted_call(f, args, kwargs, caller_fn_scope, options)
    552           'Cause: %s', target_entity, e)
    553     else:
--> 554       logging.warn(
    555           'AutoGraph could not transform %s and will run it as-is.\n'
    556           'Please report this to the TensorFlow team. When filing the bug, set'

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/autograph/utils/ag_logging.py in warn(msg, *args, **kwargs)
    144 
    145 def warn(msg, *args, **kwargs):
--> 146   logging.warn(msg, *args, **kwargs)
    147   if echo_log_to_stdout:
    148     _output_to_stdout('WARNING: ' + msg, *args, **kwargs)

~/tf38/lib/python3.8/site-packages/tensorflow_core/python/platform/tf_logging.py in warn(msg, *args, **kwargs)
    159 @tf_export(v1=['logging.warn'])
    160 def warn(msg, *args, **kwargs):
--> 161   get_logger().warning(msg, *args, **kwargs)
    162 
    163 

/usr/local/lib/python3.8/logging/__init__.py in warning(self, msg, *args, **kwargs)
   1444         """
   1445         if self.isEnabledFor(WARNING):
-> 1446             self._log(WARNING, msg, args, **kwargs)
   1447 
   1448     def warn(self, msg, *args, **kwargs):

/usr/local/lib/python3.8/logging/__init__.py in _log(self, level, msg, args, exc_info, extra, stack_info, stacklevel)
   1563             #IronPython can use logging.
   1564             try:
-> 1565                 fn, lno, func, sinfo = self.findCaller(stack_info, stacklevel)
   1566             except ValueError: # pragma: no cover
   1567                 fn, lno, func = "(unknown file)", 0, "(unknown function)"

TypeError: _logger_find_caller() takes from 0 to 1 positional arguments but 2 were given

**Describe the expected behavior**

There should be no error.  It works fine with TF2.0.0 and Python 3.6 or Python 3.7.

**Code to reproduce the issue**
Provide a reproducible test case that is the bare minimum necessary to generate the problem.

import tensorflow as tf
import numpy as np
import gzip
import json
from sklearn.model_selection import ShuffleSplit

with gzip.open("small_data/cal_house.json.gz", "r") as fin:
    housing = json.load(fin)
    
for train, test in ShuffleSplit(1, 0.2, random_state=42).split(housing['data']):
    X_train = np.array(housing['data'])[train].astype(np.float32)
    y_train = np.array(housing['target'])[train].astype(np.float32)
    X_test = np.array(housing['data'])[test].astype(np.float32)
    y_test = np.array(housing['target'])[test].astype(np.float32)

X_mean = X_train.mean(axis=0)
X_std = X_train.std(axis=0)

Xs_train = (X_train - X_mean) / X_std
Xs_test = (X_test - X_mean) / X_std

class LinearRegressionTF():
    def __init__(self, eta=.1):
        self.W = tf.Variable(0.)
        self.b = tf.Variable(0.)
        self.opt = tf.keras.optimizers.SGD(learning_rate=eta)
    
    def loss(self, X, y, return_func=False):
        def loss_():
            return tf.reduce_mean(tf.square(X * self.W + self.b - y))
        
        if not return_func:
            return loss_()
        
        return loss_

    @tf.function
    def fit(self, X, y, steps=1):
        for _ in range(steps):
            self.opt.minimize(self.loss(X, y, return_func=True), [self.W, self.b])

tf_model = LinearRegressionTF()

tf_model.fit(Xs_train[:, 0:1], y_train.reshape(-1, 1));

[cal_house.json.gz](https://github.com/tensorflow/tensorflow/files/3780890/cal_house.json.gz)

**Other info / logs**
Include any logs or source code that would be helpful to diagnose the problem. If including tracebacks, please include the full traceback. Large logs and files should be attached.
Nil

--- a/tensorflow/python/platform/tf_logging.py
+++ b/tensorflow/python/platform/tf_logging.py
@@ -57,7 +57,8 @@ def _get_caller(offset=3):
     f = f.f_back
   return None, None
 
-# The definition of `findCaller` changed in Python 3.2
+# The definition of `findCaller` changed in Python 3.2,
+# and further changed in Python 3.8
 if _sys.version_info.major >= 3 and _sys.version_info.minor >= 8:
   def _logger_find_caller(stack_info=False, stacklevel=1):  # pylint: disable=g-wrong-blank-lines
     code, frame = _get_caller(4)

--- a/tensorflow/python/platform/tf_logging.py
+++ b/tensorflow/python/platform/tf_logging.py
@@ -57,9 +57,18 @@ def _get_caller(offset=3):
     f = f.f_back
   return None, None
 
-
 # The definition of `findCaller` changed in Python 3.2
-if _sys.version_info.major >= 3 and _sys.version_info.minor >= 2:
+if _sys.version_info.major >= 3 and _sys.version_info.minor >= 8:
+  def _logger_find_caller(stack_info=False, stacklevel=1):  # pylint: disable=g-wrong-blank-lines
+    code, frame = _get_caller(4)
+    sinfo = None
+    if stack_info:
+      sinfo = '\n'.join(_traceback.format_stack())
+    if code:
+      return (code.co_filename, frame.f_lineno, code.co_name, sinfo)
+    else:
+      return '(unknown file)', 0, '(unknown function)', sinfo
+elif _sys.version_info.major >= 3 and _sys.version_info.minor >= 2:
   def _logger_find_caller(stack_info=False):  # pylint: disable=g-wrong-blank-lines
     code, frame = _get_caller(4)
     sinfo = None

